### PR TITLE
9246 SP/GP people and roles (WIP)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#9246

*Description of changes:*

**This is a partial commit for People and Roles but it includes more changes / cleanup / layout and some other fixes, so it would be nice to get this merged even now (after successful review).**

Commit 1:
- app version = 3.5.0

Commit 2:
- renamed OrgPerson -> AddEditOrgPerson
- created AddEditBusCorp + unit tests (WIP)
- styled Delete button in ListNameTranslations
- styled Delete button in ListPeopleAndRoles
- add new things to PeopleAndRoles
- moved getters and some methods from PeopleAndRoles to a new mixin
- fixed misc whitespace
- updated message and button styles in ConfirmDialog
- updated resources
- updated unit tests

Commit 3:
- updated Entity Info description re: DBA
- added more P&R blurbs for SP
- updated P&R resources
- refactored Help Section
- saved/restored reg Business Type
- added transitions to help panels
- misc cleanup

Commit 4:
- moved more things out of PeopleAndRoles and into mixin
- created RegPeopleAndRoles
- updated company description on step 3 (e: DBA

Commit 5:
- created RegAddEditOrgPerson
- moved Summary into IncorporationReviewConfirm
- changed font-weight 700 to bold
- added transitions around help blocks
- fixed some whitespace
- fixed name translation not triggering Have Changes warning

Commit 6:
- update Entity Type on step 1 re: DBA
- updated actions bg color
- changed bottom padding to top padding for responsive layout
- more work on reg P&R components
- fixed document delivery contact label
- misc cleanup

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "business-create-ui",
-  "version": "3.4.9",
+  "version": "3.5.0",
```
Comment (severinbeauvais) [inline package.json:3]: If @shabeeb-aot 's PR is merged first then this is fine.

If my PR is merged first then Faiz will have to update his (perhaps go to 3.6.0?).

File: src/components/common/AddEditBusCorp.vue [hunk 2]
```diff
@@ -0,0 +1,663 @@
+<template>
```
Comment (severinbeauvais) [inline src/components/common/AddEditBusCorp.vue:1]: New file. I copied this from AddEditOrgPerson. I still have a ton of work to do in here.

File: src/components/common/HelpSection.vue [hunk 3]
```diff
@@ -1,32 +1,26 @@
 <template>
-  <div v-if="helpSection">
```
Comment (severinbeauvais) [inline src/components/common/HelpSection.vue:2]: This component shouldn't decide whether to display itself; the parent should decide.

File: src/components/common/HelpSection.vue [hunk 4]
```diff
@@ -1,32 +1,26 @@
...
-          {{item}}
-        </li>
-      </ul>
-      <u class="help-btn" @click="helpToggle = !helpToggle"><small>Hide Help</small></u>
-    </section>
+        <p v-for="(item, index) in items" :key="index" class="my-4" v-html="item" />
+
+        <div class="help-btn bottom" @click="helpToggle = !helpToggle">Hide Help</div>
+      </section>
+    </v-expand-transition>
```
Comment (severinbeauvais) [inline src/components/common/HelpSection.vue:19]: I started generalizing this to use in other components, but those other components had too many special cases. It's still cleaner than before though :)

File: src/components/common/ListPeopleAndRoles.vue [hunk 5]
```diff
@@ -156,24 +169,26 @@ export default class ListPeopleAndRoles extends Mixins(CommonMixin) {
...
-   * @param filing The filing body which contains the name/title.
-   * @returns The appropriate Corporation or Person name.
-   */
-  private formatName (filing: any): string {
-    return filing?.officer?.organizationName ? filing?.officer?.organizationName
-      : `${filing.officer.firstName} ${filing.officer.middleName || ''} ${filing.officer.lastName}`
+  /** Returns true if org-person is a business or corporation. */
+  private isBusiness (orgPerson: OrgPersonIF): boolean {
+    return (orgPerson.officer?.partyType === PartyTypes.BUSINESS)
+  }
```
Comment (severinbeauvais) [inline src/components/common/ListPeopleAndRoles.vue:185]: I still have a ton of work to do in here.

File: src/components/common/PeopleAndRoles.vue [hunk 6]
```diff
@@ -190,98 +240,15 @@ export default class PeopleAndRoles extends Vue {
   readonly NumWord = NumWord
   readonly RuleIds = RuleIds
 
-  /** The list of organizations/persons. */
```
Comment (severinbeauvais) [inline src/components/common/PeopleAndRoles.vue:193]: I moved these to a mixin.

File: src/dialogs/ConfirmDialog.vue [hunk 7]
```diff
@@ -131,3 +131,16 @@ export default class ConfirmDialog extends Vue {
...
+
+.v-card__text {
+  color: $gray7 !important;
+}
+
+// bold the primary action button
+.dialog-yes-btn {
+  font-weight: bold;
+}
+</style>
```
Comment (severinbeauvais) [inline src/dialogs/ConfirmDialog.vue:146]: Style changes per Yui.

File: src/mixins/filing-template-mixin.ts [hunk 8]
```diff
@@ -363,12 +367,6 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
         break
     }
 
-    // If this is a named IA then add Name Request Number and Approved Name.
-    if (this.isNamedBusiness) {
-      filing.registration.nameRequest.nrNumber = this.getNameRequestNumber
-      filing.registration.nameRequest.legalName = this.getApprovedName
-    }
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:370]: I believe this block is obsolete, since the draft registration filing already contains a nameRequest object with the 3 fields that we care about. @cameron-freshworks , thoughts?

The app still fetches the actual NR and then validates it, but the app NR data is not used in the registration filing object.

This feels a bit awkward but there's a reason for both to exist (outside the control of this app) so it is what it is.

File: src/mixins/people-roles-mixin.ts [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  /** Returns true if specified org/person is a director. */
+  isDirector (orgPerson: OrgPersonIF): boolean {
+    return orgPerson.roles.some(role => role.roleType === RoleTypes.DIRECTOR)
+  }
+
+  /** Helper to show the confirm dialog. */
+  private async showConfirmDialog (title: string, message: string, action: string): Promise<boolean> {
+    const confirm = await this.$refs.confirmDialog.open(title, message, {
```
Comment (severinbeauvais) [inline src/mixins/people-roles-mixin.ts:157]: A little helper here that expects `confirmDialog` to exist on "this". Since this mixin is added to the PeopleAndRoles component's context (its "this"), this works.

TLDR; we can call the confirmDialog here but it needs to be declared in the component.

File: src/resources/Dissolutions/benefitCompany.ts [hunk 10]
```diff
@@ -19,6 +19,7 @@ export const BenefitCompanyDissolutionResource: DissolutionResourceIF = {
      addresses must be located in B.C.`,
     custodianTitle: `Custodian of Records`,
     helpSection: {
+      header: 'Help with Custodian of Records',
```
Comment (severinbeauvais) [inline src/resources/Dissolutions/benefitCompany.ts:22]: Now needed because of updates to custodian code.

File: src/resources/Incorporations/cooperative.ts [hunk 11]
```diff
@@ -35,9 +35,7 @@ export const CooperativeResource: IncorporationResourceIF = {
           person to locate the Director.`
       ]
     },
-    addIncorporator: false,
-    addOrganization: false,
-    addProprietor: false,
+    addOrganization: 'Add Organization',
```
Comment (severinbeauvais) [inline src/resources/Incorporations/cooperative.ts:38]: It's now a string so that Cooperatives can have a different label from the other entity types. This removes the "isTypeCoop" logic from the PeopleAndRoles component.

Empty string is false, of course.

File: src/store/getters/state-getters.ts [hunk 12]
```diff
@@ -188,6 +183,11 @@ export const isPremiumAccount = (state: StateIF): boolean => {
   return (getAccountInformation(state).accountType === AccountTypes.PREMIUM)
 }
 
+/** The Org Information object. */
```
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:186]: I just moved this after all the "isXxx" getters.

File: tests/unit/AddEditBusCorp.ts [hunk 13]
```diff
@@ -4,7 +4,7 @@ import Vuelidate from 'vuelidate'
 import { mount, Wrapper, createLocalVue } from '@vue/test-utils'
 import flushPromises from 'flush-promises'
 import { getVuexStore } from '@/store'
-import OrgPerson from '@/components/common/OrgPerson.vue'
+import AddEditBusCorp from '@/components/common/AddEditBusCorp.vue'
```
Comment (severinbeauvais) [inline tests/unit/AddEditBusCorp.ts:7]: GitHub diff is confused. This file is actually the new test file (cloned from OrgPerson). The next file below is the renamed one.

**Review by cameron-freshworks — APPROVED** [review]
It looks good to me. 
I'm glad you were able to shim the top layer for registrations and reuse the table, i imagine this is quite the undertaking.

**Review by vysakh-menon-aot — APPROVED** [review]
